templates/demo/app/components/blockchain.js

```diff
@@ -0,0 +1,96 @@
+import EmbarkJS from 'Embark/EmbarkJS';
+import SimpleStorage from 'Embark/contracts/SimpleStorage';
+import React from 'react';
+import { Form, FormGroup, FormControl, HelpBlock, Button } from 'react-bootstrap';
+ 
+class Blockchain extends React.Component {
+
+    constructor(props) {
+      super(props);
+  
+      this.setValue = this.setValue.bind(this);
+      this.getValue = this.getValue.bind(this);
+      this.handleChange = this.handleChange.bind(this);
```

Comment:
Just an FYI, you don't need to premptively bind your functions if you call them with arrows. Eg:
```
onClick={(e) => this.setValue(e)}
```
or
```
 handleChange = (e) => {
  this.setState({valueSet: e.target.value});
};
```

Reply:
Updated the components based on this suggestion